javagen/src/main/java/com/azure/autorest/mapper/ClientMethodMapper.java

```diff
@@ -684,6 +722,56 @@ private String getPageableItemName(Operation operation) {
             .map(ClientModelProperty::getName).findAny().orElse(null);
     }
 
+    private IType getPollingIntermediateType(JavaSettings.PollingDetails details, IType syncReturnType) {
+        IType pollResponseType = syncReturnType.asNullable();
+        if (JavaSettings.getInstance().isFluent()) {
+            return pollResponseType;
+        }
+        if (details != null && details.getIntermediateType() != null) {
+            String intermediateTypeName;
+            String intermediateTypePackage;
+            if (details.getIntermediateType().contains(".")) {
+                intermediateTypeName = details.getIntermediateType().replaceAll(".*\\.", "");
+                intermediateTypePackage = details.getIntermediateType().replace("." + intermediateTypeName, "");
+            } else {
+                intermediateTypeName = details.getIntermediateType();
+                intermediateTypePackage = JavaSettings.getInstance().getPackage();
+            }
+            pollResponseType = new ClassType.Builder().packageName(intermediateTypePackage).name(intermediateTypeName).build();
+        }
+        // azure-core wants poll response to be non-null
+        if (pollResponseType == ClassType.Void) {
+            pollResponseType = ClassType.BinaryData;
+        }
```

Comment:
Good point. Mgmt might able to adopt this as well.

Reply:
I think in Mgmt you have your own `PollResult` type - I actually am envious of that. In data plane we do not have a common data structure all services agreed upon, so we have to use a `BinaryData` here. But if all services have a common type (like in mgmt), I think having a strong type may be more helpful.

Reply:
`PollResult<BinaryData>` would be better than `PollResult<Void>`.